src/main/java/com/solvd/pages/AbstractPage.java

```diff
@@ -0,0 +1,68 @@
+package com.solvd.pages;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.openqa.selenium.TimeoutException;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public abstract class AbstractPage {
+    private static final Logger LOGGER = LogManager.getLogger(AbstractPage.class);
+    private WebDriver driver;
+
+    public AbstractPage(WebDriver driver){
+        this.driver = driver;
+        PageFactory.initElements(driver, this);
+    }
+
+    public void click(WebElement element){
+        try{
+            waitUntilElementToBeClickable(element);
+            element.click();
+            LOGGER.info("Element with id {} was clicked", element.getAttribute("id"));
```

Comment:
not every element that you will click will have "id" as attribute, so logger info may not be true when element without id is clicked.

Reply:
yes, I agree. it only worked at this stage. I can't leave it like this